File: 7795_먹을 것인가 먹힐 것인가.cpp

//풀이
//정렬, 두 포인터

#include<iostream>
#include<algorithm>
#include<vector>
using namespace std;
bool desc(int a, int b)
{
	return a > b;
}
int main()
{
	int T;
	cin >> T;
	while (T--) {
		int N, M;
		vector<int> n;
		vector<int> m;
		cin >> N >> M;

		for (int i = 0; i < N; i++) {
			int num;
			cin >> num;
			n.push_back(num);
		}
		for(int i = 0; i < M; i++) {
			int num;
			cin >> num;
			m.push_back(num);
		}

		sort(n.begin(), n.end(), desc);
		sort(m.begin(), m.end(), desc);

		int num = 0, cnt = 0;
		for (int i = 0; i < n.size(); i++) {
			for (int j = num; j < m.size(); j++) {
				if (n[i] > m[j]) {
					cnt += m.size() - j;
					num = j;
					break;
				}
			}
		}
		cout << cnt << "\n";
	}
}

File: 7795_먹을 것인가 먹힐 것인가.java

//풀이
//정렬, 두 포인터

import java.util.*;

public class Main {

    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);
        int T = scan.nextInt();
        while(T > 0){
            T--;
            int N = scan.nextInt();
            int M = scan.nextInt();
            ArrayList<Integer> n = new ArrayList<Integer>();
            ArrayList<Integer> m = new ArrayList<Integer>();
            for(int i = 0; i<N; i++){
                int num = scan.nextInt();
                n.add(num);
            }
            for(int i = 0; i<M; i++){
                int num = scan.nextInt();
                m.add(num);
            }
            Collections.sort(n, new Desc());
            Collections.sort(m, new Desc());
            int cnt = 0, num = 0;
            for(int i = 0; i<n.size(); i++){
                for(int j = num; j<m.size(); j++){
                    if(n.get(i) > m.get(j)) {
                        num = j;
                        cnt += m.size() - j;
                        break;
                    }
                }
            }
            System.out.println(cnt);
        }
    }

    static class Desc implements Comparator<Integer> {
        @Override
        public int compare(Integer a, Integer b){
            return b.compareTo(a);
        }
    }
}
